grails-app/services/au/org/ala/phyloviz/TreeService.groovy

```diff
@@ -605,4 +607,31 @@ class TreeService {
         }
         return result;
     }
+
+    List getSpeciesNamesFromTree(Integer treeId) {
+        JadeTree tree = metricsService.getJadeTree(metricsService.treeProcessing(Tree.findById(treeId).tree))
+
+        metricsService.getLeafNames(tree)
+    }
+
+    /**
+     * Removes all character lists that do no contain any of the species in the specified tree
+     *
+     * @param treeId The ID of the tree to use for filtering
+     * @param characterLists Collection of character lists to be filtered
+     * @return Subset of the characterLists where at least 1 species in the tree is in each list
+     */
+    List filterCharacterListsByTree(Integer treeId, Collection characterLists) {
+        List speciesNames = getSpeciesNamesFromTree(treeId)
+
+        Map request = [scientificNames: speciesNames, drIds: characterLists.collect { it.dataResourceId }]
+
+        String url = "${grailsApplication.config.listToolBaseURL}/ws/speciesList/filter"
```

Comment:
this variable is not defined in config.groovy. is it defined in external config file?